File: Java/CesiumLanguageWriter/translatedSrc/cesiumlanguagewriter/advanced/CesiumCartographicRectangleRadiansValuePropertyAdaptor.java
package cesiumlanguagewriter.advanced;


import agi.foundation.compatibility.*;
import cesiumlanguagewriter.*;
import javax.annotation.Nonnull;

/**
 *  
 Adapts a class that implements {@link ICesiumCartographicRectangleRadiansValuePropertyWriter} to implement
 {@link ICesiumInterpolatableValuePropertyWriter} for {@link CartographicExtent} values.
 
 

 * @param <TFrom> The class that implements {@link ICesiumCartographicRectangleRadiansValuePropertyWriter} to adapt.
 */
@SuppressWarnings( {
        "unused",
        "deprecation",
        "serial"
})
public class CesiumCartographicRectangleRadiansValuePropertyAdaptor<TFrom extends ICesiumCartographicRectangleRadiansValuePropertyWriter> extends
        CesiumInterpolatableWriterAdaptor<TFrom, CartographicExtent> {
    /**
    *  
    Initializes a new instance.
    
    
    
    
    

    * @param parent The instance to wrap.
    * @param writeValueCallback The callback to write values of type {@link CartographicExtent}.
    * @param writeSamplesCallback The callback to write samples of type {@link CartographicExtent}.
    * @param writeDeleteValueCallback The callback to write an indication that the client should delete existing data.
    */
    public CesiumCartographicRectangleRadiansValuePropertyAdaptor(@Nonnull TFrom parent, @Nonnull CesiumWriterAdaptorWriteCallback<TFrom, CartographicExtent> writeValueCallback,
            @Nonnull CesiumWriterAdaptorWriteSamplesCallback<TFrom, CartographicExtent> writeSamplesCallback, @Nonnull CesiumWriterAdaptorWriteDeleteCallback<TFrom> writeDeleteValueCallback) {
        super(parent, writeValueCallback, writeSamplesCallback, writeDeleteValueCallback);
    }
}

File: DotNet/CesiumLanguageWriter/Advanced/CesiumCartographicRectangleRadiansValuePropertyAdaptor.cs
using JetBrains.Annotations;

namespace CesiumLanguageWriter.Advanced
{
    /// <summary>
    /// Adapts a class that implements <see cref="ICesiumCartographicRectangleRadiansValuePropertyWriter"/> to implement
    /// <see cref="ICesiumInterpolatableValuePropertyWriter{TValue}"/> for <see cref="CartographicExtent"/> values.
    /// </summary>
    /// <typeparam name="TFrom">The class that implements <see cref="ICesiumCartographicRectangleRadiansValuePropertyWriter"/> to adapt.</typeparam>
    public class CesiumCartographicRectangleRadiansValuePropertyAdaptor<TFrom> : CesiumInterpolatableWriterAdaptor<TFrom, CartographicExtent>
        where TFrom : class, ICesiumCartographicRectangleRadiansValuePropertyWriter
    {
        /// <summary>
        /// Initializes a new instance.
        /// </summary>
        /// <param name="parent">The instance to wrap.</param>
        /// <param name="writeValueCallback">The callback to write values of type <see cref="CartographicExtent"/>.</param>
        /// <param name="writeSamplesCallback">The callback to write samples of type <see cref="CartographicExtent"/>.</param>
        /// <param name="writeDeleteValueCallback">The callback to write an indication that the client should delete existing data.</param>
        public CesiumCartographicRectangleRadiansValuePropertyAdaptor([NotNull] TFrom parent,
                                                                      [NotNull] CesiumWriterAdaptorWriteCallback<TFrom, CartographicExtent> writeValueCallback,
                                                                      [NotNull] CesiumWriterAdaptorWriteSamplesCallback<TFrom, CartographicExtent> writeSamplesCallback,
                                                                      [NotNull] CesiumWriterAdaptorWriteDeleteCallback<TFrom> writeDeleteValueCallback)
            : base(parent, writeValueCallback, writeSamplesCallback, writeDeleteValueCallback)
        {
        }
    }
}